### PR TITLE
Config module tests

### PR DESCRIPTION
**Pull request by SatyendraBanjare**
Added Config and Buf Module Tests. :smile:

### CODE REVIEW

File: src/luagit2/config/lua_config_help.c [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+// 	/** Application specific configuration file; freely defined by applications
+// 	 */
+// 	GIT_CONFIG_LEVEL_APP = 6,
+
+// 	/** Represents the highest level available config file (i.e. the most
+// 	 * specific config file available that actually is loaded)
+// 	 */
+// 	GIT_CONFIG_HIGHEST_LEVEL = -1,
+// } git_config_level_t;
```
Comment (Etiene) [inline src/luagit2/config/lua_config_help.c]: Why was this commented out?

File: tests/blob/test_blob_spec.lua [hunk 2]
```diff
@@ -1,7 +1,7 @@
 -- Tests for luagit2's Blob module functions
 
 local fixer = require("Fixtures.fix_repo")
-local lfs = require("lfs")
+
```
Comment (Etiene) [inline tests/blob/test_blob_spec.lua:4]: why is this removed here?

Comment (SatyendraBanjare) [reply tests/blob/test_blob_spec.lua:4]: There was no use of lfs module to check the path in this test. Therefore I removed it.

File: src/luagit2/config/lua_config_help.c (hunk 1 above)
Comment (SatyendraBanjare) [reply src/luagit2/config/lua_config_help.c]: That is my mistake, I used to refer it while creating bindings. It has absolutely nothing to do with any of code. I will remove it. Thankyou

File: tests/buf/test_buf_spec.lua [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	-- When you buf details method is called
+	-- The git_buf object Buf's (Buf.ptr) is cleared to NULL
+	-- Hence For now, It causes error on using buf_free
+	-- This Also means You cannot cal luagit2_buf_details()
+	-- Again and Again on same luagit2_buffer Object
+	--
+	--
+	it("Tests Buf Free Method",function()
+		local buffer = lib.luagit2_buf_set_str("Hello")
+		--lib.luagit2_buf_free(buffer)
```
Comment (Etiene) [inline tests/buf/test_buf_spec.lua]: This test isnt doing anything! Please adjust :)

File: tests/config/test_config_spec.lua [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+		sys_email = sys_email_handle:read("*a")
+		sys_email_handle:close()
+
+		local global_name_handle = io.popen("git config --global --get user.name ")
+		global_name = global_name_handle:read("*a")
+		global_name_handle:close()
+
+		local global_email_handle = io.popen("git config --global --get user.email ")
+		global_email = global_email_handle:read("*a")
+		global_email_handle:close()
```
Comment (Etiene) [inline tests/config/test_config_spec.lua]: It looks like there is a lot of repetition here. One way to improve it write a local function at the top of test file, just before the first describe that encapsulates opening a command, reading and closing it, then you just pass the command string here :)

File: tests/config/test_config_spec.lua [hunk 5]
```diff
@@ -0,0 +1,363 @@
...
+	end)
+
+	-- it("Tests config open System ", function()
+	-- 	Config_Snapshot = lib.luagit2_config_snapshot(Config_System)
+	-- 	local name = lib.luagit2_config_get_string(Config_Snapshot,"user.name")
+	-- 	local email = lib.luagit2_config_get_string(Config_Snapshot,"user.email")
+
+	-- 	assert.are.equal(name .. '\n', sys_name)
+	-- 	assert.are.equal(email .. '\n',sys_email)
+	-- end)
```
Comment (Etiene) [inline tests/config/test_config_spec.lua]: Why is this test commented out?

File: tests/buf/test_buf_spec.lua (hunk 3 above)
Comment (SatyendraBanjare) [reply tests/buf/test_buf_spec.lua]: I have fixed it in a later commit.

File: tests/config/test_config_spec.lua (hunk 4 above)
Comment (SatyendraBanjare) [reply tests/config/test_config_spec.lua]: ok thankyou. I will try to do so.

File: tests/config/test_config_spec.lua (hunk 5 above)
Comment (SatyendraBanjare) [reply tests/config/test_config_spec.lua]: this function `luagit2_config_open_system()` was giving errors. so I commented that part out.

Comment (SatyendraBanjare) [reply tests/config/test_config_spec.lua]: I will try to fix it.

File: tests/blob/test_blob_spec.lua (hunk 2 above)
Comment (Etiene) [reply tests/blob/test_blob_spec.lua:4]: ok!

File: tests/buf/test_buf_spec.lua (hunk 3 above)
Comment (Etiene) [reply tests/buf/test_buf_spec.lua]: can you post the link here so I have a look?

File: tests/config/test_config_spec.lua (hunk 5 above)
Comment (Etiene) [reply tests/config/test_config_spec.lua]: ok! :)  remember to add the fixes on this PR and not a different PR :)

File: tests/buf/test_buf_spec.lua (hunk 3 above)
Comment (SatyendraBanjare) [reply tests/buf/test_buf_spec.lua]: https://github.com/SatyendraBanjare/luagit2/blob/0e1049311557a8895b9fa1686b14b22cc96df14a/tests/buf/test_buf_spec.lua#L14

**Comment by SatyendraBanjare** [comment]
@Etiene  mam, I have fixed those few small errors. I have removed those unnecessary comments. Also I realized there is no extra function to open system level config, instead we can do that using `open_config_level()` and then passing system as the option.

File: tests/config/test_config_spec.lua [hunk 6]
```diff
@@ -0,0 +1,345 @@
+-- Luagit2's Config Module Tests
+
+local fixer = require("Fixtures.fix_repo")
+local lfs = require("lfs")
+
+local function get_config_details(option_string)
+	name_handle = io.popen("git config --get --" .. option_string .." user.name ")
```
Comment (Etiene) [inline tests/config/test_config_spec.lua]: global variable here! are you using luacheck?

**Comment by SatyendraBanjare** [comment]
@Etiene  mam, I re installed my editor, forgot to install linter. Fixed the error now. :smile:

**Review by Etiene — APPROVED** [review]
👍
